packages/inter-protocol/src/priceOracleAdmin.js

```diff
@@ -0,0 +1,95 @@
+import { defineDurableFarClass, M, makeKindHandle } from '@agoric/vat-data';
+
+export const INVITATION_MAKERS_DESC = 'oracle invitation';
+
+/**
+ * @typedef {{
+ *   endingRound: bigint,
+ *   oracleId: string,
+ *   roundPowers: { handlePush: Function }
+ * }} HeldParams
+ */
+
+/**
+ * @typedef {{ roundId: number | undefined, unitPrice: NatValue }} PriceDatum
+ */
+
+/**
+ * @typedef {object} OracleStatus
+ * @property {bigint} endingRound
+ * @property {bigint} lastReportedRound
+ * @property {bigint} lastStartedRound
+ * @property {bigint} latestSubmission
+ */
+/**
+ * @typedef {Readonly<HeldParams & {
+ * }>} ImmutableState
+ *
+ * @typedef {OracleStatus & {
+ * }} MutableState
+ */
+/** @typedef {ImmutableState & MutableState} State */
+
+const oracleAdminKind = makeKindHandle('OracleAdmin');
+
+/**
+ * @param {HeldParams} heldParams
+ * @returns {State}
+ */
+const initState = heldParams => {
+  return {
+    ...heldParams,
+    lastReportedRound: 0n,
+    lastStartedRound: 0n,
+    latestSubmission: 0n,
+  };
+};
```

Comment:
> This allows the caller of makeOracleAdmin() to determine the fields that will be recorded in state. I'd rather that list were statically determined here. 

Good point. Done.

Reply:
btw, what do you think of this other approach:
https://github.com/Agoric/agoric-sdk/blob/805326e062fb162a6ee6fb93ca76e9a53299cd98/packages/smart-wallet/src/smartWallet.js#L202

I think I prefer it because,
- like the explicit keys it prevents additional values from being injected
- on top of that it can validate the shapes
- it keeps the values logically organized

Reply:
so many declarations of `M.eref(M.any())` hurts my eyes. 

I've taken to declaring constants for those opaque types in hopes that we can do better over time. It's at least more informative to read with names, even when it can't enforce any more.
```
export const BankShape = M.remotable('Bank');
...

  fit(
    unique,
    harden({
      address: M.string(),
      bank: M.eref(BankShape),
      invitationPurse: M.eref(InvitationShape),
    }),
  );
```
 (InvitationShape exists.)

But yeah, it's nice to be able to use patterns this way.

Reply:
Note that `M.eref(M.any())` is operationally equivalent to `M.any()`.

`M.eref(M.remotable(...))` is meaningful and irreducible. No simpler pattern means the same thing.

Reply:
OK, I'll bite; how would we observe the difference between `M.any()` and `M.eref(M.remotable(...))` in a synchronous call to `fit`?

Reply:
I have not tested the following claims:

`fit(8, M.any())` passes

`fit(8, M.eref(M.any()))` passes

`fit(8, M.eref(M.remotable('foo')))` fails, since `8` is neither a promise nor a remotable.